src/test/java/cloud/prefab/client/integration/IntegrationTestFile.java

```diff
@@ -0,0 +1,48 @@
+package cloud.prefab.client.integration;
+
+import cloud.prefab.client.integration.DynamicTestBuilder;
+import java.util.ArrayList;
+import java.util.Map;
+import org.junit.jupiter.api.DynamicTest;
+
+public class IntegrationTestFile {
+
+  String version;
+  String function;
+  ArrayList<Map<String, Object>> tests;
+
+  public IntegrationTestFile(
+    String version,
+    String function,
+    ArrayList<Map<String, Object>> tests
```

Comment:
List<Map<String, Object>> tests.   We just generally want to use the least specific one necessary.